plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/registry/formatter/DataFormatterProfile.java

```
@@ -199,21 +203,93 @@ public DBDDataFormatter createFormatter(@NotNull String typeId, DBSTypedObject t
         if (descriptor == null) {
             throw new IllegalArgumentException("Formatter '" + typeId + "' not found");
         }
-        DBDDataFormatter formatter = descriptor.createFormatter();
 
         Map<String, Object> defProps = descriptor.getSample().getDefaultProperties(locale);
         Map<String, Object> props = getFormatterProperties(store, typeId);
         Map<String, Object> formatterProps = new HashMap<>();
         if (defProps != null && !defProps.isEmpty()) {
             formatterProps.putAll(defProps);
         }
-        if (props != null && !props.isEmpty()) {
+        if (!props.isEmpty()) {
             formatterProps.putAll(props);
         }
+
+        if ("number".equalsIgnoreCase(typeId)) {
+            boolean excludeIds = CommonUtils.toBoolean(
+                formatterProps.getOrDefault(NumberFormatSample.PROP_EXCLUDE_ID_COLUMNS, false)
+            );
+            if (excludeIds && type instanceof DBSTableColumn column && (isPrimaryKey(column) || isForeignKey(column))) {
+                formatterProps.put(NumberFormatSample.PROP_USE_GROUPING, false);
+            }
+        }
+
+        DBDDataFormatter formatter = descriptor.createFormatter();
         formatter.init(type, locale, formatterProps);
+
         return formatter;
     }
 
+    static boolean isPrimaryKey(DBSTableColumn column) {
+        var monitor = new VoidProgressMonitor();
+        try {
+            return matchesColumnInRefs(
+                column,
+                referrers(column.getParentObject().getConstraints(monitor), DBSEntityConstraintType.PRIMARY_KEY),
+                monitor
+            );
+        } catch (DBException e) {
```

Do not ignore exception, use log.debug